source/elements/oneDPL/source/parallel_api/iterators.rst

```diff
@@ -3,9 +3,53 @@
 ..
 .. SPDX-License-Identifier: CC-BY-4.0
 
+.. _iterators:
+
 Iterators
 ---------
 
+Requirements For Iterator Use With Device Policies
+++++++++++++++++++++++++++++++++++++++++++++++++++
+Iterators and iterator-like types may or may not refer to content accessible on the device. The term *indirectly device
+accessible* refers to a type that represents content accessible on the device within a `SYCL`_ kernel. *Indirectly
+device accessible iterators* are iterators that can inherently be dereferenced on the device within a SYCL kernel.
+
+Examples of *indirectly device accessible iterators* include SYCL USM shared or device memory pointers, as well as
+iterator types like ``counting_iterator`` or ``discard_iterator``, which can inherently be used in a SYCL kernel. An
+example of an iterator type that is not *indirectly device accessible* is a ``std::vector`` iterator with a host
+allocator, as it cannot inherently be used in a SYCL kernel.
```

Comment:
These examples seem mostly redundant, because the same information is provided below. Perhaps one example would be enough to illustrate the introduced terms.

Reply:
I can reduce the number mentioned here, but I think there are probably 4 important categories to call out in this introduction, and I think all are useful to introduce the concept. 

1) iterator with no data (generates on the fly)
2) iterator with device accessible data
3) iterator with only host accessible data
4) buffer wrapper (indirectly device accessible but not iterator)

I could be convinced to drop (1), since we have those on this same page.

Reply:
I tend to think these examples belongs to the developer guide rather than the specification.
(4) is OK as a requirement, not an example. (3) I would also drop, or perhaps combine with (2) - for example, `std::vector` with a USM allocator has an iterator that is indirectly device accessible, bit not with a host allocator.

Reply:
Though maybe `std::vector` with USM allocator is bad for an example, as we do not want to have an impression that it is guaranteed to work efficiently.

Reply:
I cut it down to one positive and one negative example.  If we want to just remove this paragraph we can, and just rely upon the definition and rules below.